### DISCUSSION TITLE
Local checks

### DISCUSSION THREAD

**Discussion by Marcuccio** [projectdiscovery/nuclei-templates#8920] [Ideas]
Greetings everyone,

With the introduction of version 3 and the JavaScript engine, I've been contemplating the myriad of local checks that can now be executed. Do you envision the possibility of having a designated Folder or Tag labeled "Local" to distinguish templates that aren't utilized for Remote Checks?

**Comment by olearycrew** [comment]
I think that would make a lot of sense - either a folder or a tag or both. Maybe @princechaddha could weigh in here?

**Comment by princechaddha** [comment]
@Marcuccio, we have code templates with local exploits that are not run by default. Users need to provide the -code flag, and these are marked with 'privesc' and 'local' tags. You can view them using `nuclei -tags local -vv -code`.

**Reply by Marcuccio** [reply]
I see, good to know, thanks.
Don't you think that also a dedicated folder `local` would improve readability and help testers to properly see what local templates are available?

**Reply by princechaddha** [reply]
@Marcuccio In the templates project, we have organized templates based on the protocol they use, making it easier to navigate and write templates for a specific protocol. While we do tag local exploits with a 'local' and 'privesc' tag, we can't create a separate folder for them, as most local exploits are also CVEs and thus reside in the CVE folder of their respective directory. Creating a separate 'local' folder would disrupt the current structure of the template directories. Please let me know if you have any other ideas on how we can improve this without changing the current structure.

**Reply by Marcuccio** [reply]
If I can suggest somethhing I would add the  tag local in the default ignore list as most of the times people use nuclei for remote scanning.

**Reply by princechaddha** [reply]
@Marcuccio, thank you for the suggestion. We have added the 'local' tag in the Nuclei ignore list. You can find it here: https://github.com/projectdiscovery/nuclei-templates/pull/9039
